server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -263,25 +239,33 @@ private void allocateUnassigned() {
             }
 
             /*
+             * Create some comparators to sort the unassigned shard copies in prioritized allocation order.
              * TODO: We could be smarter here and group the shards by index and then
              * use the sorter to save some iterations.
              */
             final PriorityComparator secondaryComparator = PriorityComparator.getAllocationComparator(allocation);
-            final Comparator<ShardRouting> comparator = (o1, o2) -> {
+            final Comparator<ShardRouting> primaryComparator = (o1, o2) -> {
```

Comment:
I think I find this renaming a bit confusing because `primary` ends up being overloaded a bit in this context.

We sort the arrays so that primary shards are first in the array, but the `primary` comparator does more than just that and also delegates to the `secondary` comparator.

i.e. "primary" in this context refers to the shard routing (which shard is primary) and the first of two priorities of sorting. If I'm understanding it correctly.

It's a nitpick but I wonder if we're muddying the waters a little?

Reply:
perhaps it could be "comparator" and "allocationComparator" or some descriptive name about what the second comparison is?

Reply:
It's also possible I just don't understand this code, because there's also a primary and secondary array.

Reply:
Changed primary and secondary array names to get rid of that confusion, per [other thread](https://github.com/elastic/elasticsearch/pull/121771/files#r1950085720).

The second comparator appears to focus on index comparison, so I've renamed it `indexPriorityComparator`. I've replaced primary with `shardAllocationPriorityComparator` to be clear about what it does.